### PR TITLE
Convert Java projects to Maven

### PR DESCRIPTION
**Pull request by mikepthomas**
Converted libtiled-java and tmxviewer to maven projects
Upgraded to Java 7
Deprecated use of old classes and methods
Add support for multi linked images in Tilesets
Auto generate licenses and missing JavaDocs

### CODE REVIEW

**Comment by Ablu** [comment]
Apparently the CI checks broke. I think the Java parts are supposed to be build by Qbs now...

It seems to have some issues to find some libraries:
~~~
ERROR: Error: Process 'C:\Progra~1\Java\jdk1.8.0/bin/java' finished with exit code 1. The standard error output was:
C:\projects\tiled\util\java\libtiled-java\src\test\java\tiled\io\MapReaderTest.java:35: error: package junit.framework does not exist
import junit.framework.TestCase;
                      ^
C:\projects\tiled\util\java\libtiled-java\src\test\java\tiled\io\MapReaderTest.java:40: error: cannot find symbol
public class MapReaderTest extends TestCase {
                                   ^
  symbol: class TestCase
C:\projects\tiled\util\java\libtiled-java\src\test\java\tiled\io\MapReaderTest.java:46: error: cannot find symbol
        assertTrue(reader.accept(new File("example.tmx")));
        ^
~~~

I do not think that Qbs supports Maven at the moment...

Was there any specific reason for the switch to Maven?

**Comment by mikepthomas** [comment]
Hi Eric, I have been using libtiled-java for a personal project written in Java using Maven for build and dependency management. I noticed you had an open issue that the library is not in the Maven central repository (https://github.com/bjorn/tiled/issues/397). The Java project would need to be converted to Maven to facilitate this.

**Comment by bjorn** [comment]
Does libtiled-java now depend on junit? Could this dependency be avoided, for example by moving MapReaderTest.java to a separate directory?

**Comment by mikepthomas** [comment]
JUnit is just used for the unit tests, I've declared it as a dependency in test scope in the Maven POM... If Maven is used for build, it will download it from Maven central. If Qbs cannot use Maven to build the JARs I can try re-instate the Ant build scripts to work alongside the new project structure if that would help?

**Comment by mikepthomas** [comment]
Unfortunately I don't know why the CI build is failing... the JARs seem to build ok when using both Maven and the Ant scripts now. I've never used Qbs before, but it looks like the junit JAR is not in the classpath when trying to build.

**Comment by bjorn** [comment]
> If Qbs cannot use Maven to build the JARs I can try re-instate the Ant build scripts to work alongside the new project structure if that would help?

Qbs is building the .jar files directly, it does not depend on the Maven or Ant build scripts. Feel free to remove the Ant script if it isn't useful to anybody anymore now that a Maven script is provided.

File: util/java/libtiled-java/src/main/java/tiled/core/Map.java [hunk 1]
```diff
@@ -304,8 +402,19 @@ public String getFilename() {
      * Returns a vector with the currently loaded tileSets.
      *
      * @return Vector
+     * @deprecated Use getMapTileSets instead.
```
Comment (bjorn) [inline util/java/libtiled-java/src/main/java/tiled/core/Map.java]: Do we really need to do this? So far there aren't any compatibility guarantees and it should be a small thing to fix for projects upgrading to a new libtiled-java if we just change the return type, right?

Comment (mikepthomas) [inline util/java/libtiled-java/src/main/java/tiled/core/Map.java]: I wasn't too sure what level of API compatability you were after, If there's no compatibility guarantees, i've removed the deprecated methods and just changed the return type as suggested, I've also removed the commits that added the Ant Scripts and changes to junit.

As for the comment for adding support for multi linked images in tilesets, I have added a new method in TMXMapWriter, "writeImage" and a new property in Tile (with getter and setter), "source" that allows to write individual tiles to a tilemap as links to external files. This creates an identical format to using Type "Collection of Images" when creating a new Tileset in the Tiled application v0.17. e.g:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<tileset name="TileSet" tilewidth="32" tileheight="32" tilecount="2123" columns="0">
 <tile id="0">
  <image width="32" height="32" source="tiles/0.png"/>
 </tile>
</tileset>
```

**Comment by bjorn** [comment]
> I've also removed the commits that added the Ant Scripts and changes to junit.

Something seems to have gone wrong here since now the history looks even stranger. There's a duplicated commit for "Update Qbs build scripts to point at new java root package" and a commit to remove the deprecated methods that is at the same time a merge commit.

Are you familiar enough with git to see about fixing this up or shall I try to nudge things in shape before merging it?

**Comment by mikepthomas** [comment]
Sorry I only rebase removed them yesterday, rewriting history of pushed commits us usually frowned upon... I've now removed the commits from the history too

**Comment by bjorn** [comment]
Thanks a lot for all the changes! Will you be able to close #397 now?

**Comment by mikepthomas** [comment]
Hi @bjorn, I should be able to get it as far as the development version repository (Snapshots) A few minor changes will need to be made to get it into central. I'm going to apply for the namespace so that it can be deployed, if you could register an account from https://issues.sonatype.org/secure/Signup!default.jspa and let me know your username so you can also have admin access, I'll get the ball rolling

**Comment by bjorn** [comment]
@mikepthomas Alright, I've signed up as `bjorn`. Thanks!

**Pull request merged**